### PR TITLE
Reworks mutation testing docs

### PR DESCRIPTION
**Pull request by nunomaduro**

### CODE REVIEW

File: mutation-testing.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+})->covers(CreateTodoRequest::class);
+```
+
+Finally, run Pest PHP with the `--mutate` option to start mutation testing.
+
+```bash
+./vendor/bin/pest --mutate
+```
+
+Optionally, you may run mutation testing in parallel to speed up the process. Of course, your test suite must be able to run in tests in parallel.
```
Comment (gehrisandro) [inline mutation-testing.md:42]:
```suggestion
Optionally, you may run mutation testing in parallel to speed up the process. Of course, your test suite must be able to run the tests in parallel.
```

File: mutation-testing.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+```bash
+./vendor/bin/pest --mutate --parallel
+```
+
+After running mutation testing, you will see a detailed report showing the number of mutations, how many were caught, and the mutation score.
```
Comment (gehrisandro) [inline mutation-testing.md:48]:
```suggestion
After running mutation testing, you will see a detailed report showing the number of mutations, how many were untested, and the mutation score.
```

File: mutation-testing.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+```
+[IMAGE HERE]
+```
+
+The higher the mutation score, the better your test suite is. A mutation score of 100% means that all mutations were "covered", which is the goal of mutation testing.
+
+Now, if you see "uncovered" mutations, or are a mutation score below 100%, typically means that you have **missing tests** or that **your tests are not covering all the edge cases**.
```
Comment (gehrisandro) [inline mutation-testing.md:56]:
```suggestion
Now, if you see "untested" mutations, or have a mutation score below 100%, typically means that you have **missing tests** or that **your tests are not covering all the edge cases**.
```

File: mutation-testing.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+<a name="covered-vs-uncovered-mutations"></a>
+## Covered Vs Uncovered Mutations
+
+When running mutation testing, you will see two types of mutations: **covered** and **uncovered** mutations.
+
+- **Covered Mutations**: These are mutations that were detected by your test suite. They are considered "covered" because your tests were able to catch the changes introduced by the mutation.
+
+As example, the following mutation is considered "covered" because the test suite was able to detect the change.
```
Comment (gehrisandro) [inline mutation-testing.md:65]:
```suggestion
<a name="tested-vs-untested-mutations"></a>
## Tested vs Untested and Uncovered Mutations

When running mutation testing, you will see three types of mutations: **tested**, **untested** and **uncovered** mutations.

- **Tested Mutations**: These are mutations that were detected by your test suite. They are considered "tested" because your tests were able to catch the changes introduced by the mutation.

As example, the following mutation is considered "tested" because the test suite was able to detect the change.
```

File: mutation-testing.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    // this fails because the mutation changed the return value, proving that the test is working and testing the return value...
+    $this->getJson('/todos')->assertStatus(200)->assertJsonContains([
+        ['name' => 'Buy milk'], 
+    ]);
+});
+```
+
+- **Uncovered Mutations**: These are mutations that were not detected by your test suite. They are considered "uncovered" because your tests were not able to catch the changes introduced by the mutation.
+
+As example, the following mutation is considered "uncovered" because the test suite was not able to detect the change.
```
Comment (gehrisandro) [inline mutation-testing.md:89]:
```suggestion
- **Untested Mutations**: These are mutations that were not detected by your test suite. They are considered "untested" because your tests were not able to catch the changes introduced by the mutation.

As example, the following mutation is considered "untested" because the test suite was not able to detect the change.
```

File: mutation-testing.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+it('list todos', function () {
+    Todo::factory()->create(['name' => 'Buy milk']);
+
+    // this test still passes even though the return value was changed by the mutation... 
+    $this->getJson('/todos')->assertStatus(200);
+});
+```
+
+Changing the return value is only one of many possible mutations. Typically, a mutation can be a change in the return value, a change in the method call, a change in the method arguments, and so on.
+
```
Comment (gehrisandro) [inline mutation-testing.md:110]:
```suggestion
- **Uncovered Mutations**: These are mutations were no test is excuting the mutated code at all. In this case it **always** means that a test is missing.

```

**Review by gehrisandro — CHANGES_REQUESTED** [review]
Hi @nunomaduro 

Some suggestions to improve the docs to match the used terminology.
